src/dlls/mscoree/unixinterface.cpp

```diff
@@ -146,6 +138,61 @@ int coreclr_initialize(
     ConstWStringArrayHolder propertyValuesW;
     propertyValuesW.Set(StringArrayToUnicode(propertyCount, propertyValues), propertyCount);
 
+    // Process properties that specify CoreCLR startup flags and remove them from the property list
+    STARTUP_FLAGS startupFlags =
+        static_cast<STARTUP_FLAGS>(
+            STARTUP_FLAGS::STARTUP_LOADER_OPTIMIZATION_SINGLE_DOMAIN |
+            STARTUP_FLAGS::STARTUP_SINGLE_APPDOMAIN);
+    int newPropertyIndex = 0;
+    for (int oldPropertyIndex = 0; oldPropertyIndex < propertyCount; ++oldPropertyIndex)
+    {
+        if (wcscmp(propertyKeysW[oldPropertyIndex], W("CONCURRENT_GC")) == 0)
+        {
+            if (wcscmp(propertyValuesW[oldPropertyIndex], W("1")) == 0)
+            {
+                startupFlags = static_cast<STARTUP_FLAGS>(startupFlags | STARTUP_CONCURRENT_GC);
```

Comment:
> I guess that in legacy hosting interface, you have to specify all the flags, no flags are on by default. I think that @jkotas meant that that flag would be on by default and the caller could pass "0" as the property value to turn it off.

I have not thought about this option - but it sounds good too if you would like to do it this way.